### ISSUE TITLE
Pilot Test & Small Scale Hyperparameter Optimization

### ISSUE THREAD

**Issue by psaegert** [MrWhatZitToYaa/IGNNS-final-project#20]
- Implement a wrapper for the CNF with a fit function using the existing model and train functions.
- Use https://scikit-optimize.github.io/stable/modules/generated/skopt.BayesSearchCV.html to perform hyperparameter optimization with fixed N and, lr, and scheduler psrameters (only vary model parameters)
- Add support to store and load bayesian optimizer to pause and resume optimization
